modules/postprocessing/src/processors/depthoffield.cpp

```diff
@@ -0,0 +1,427 @@
+    size2_t clickPos(mouseEvent->x(), mouseEvent->y());
+    double depthNdc =
+        inport_.getData()->getDepthLayer()->getRepresentation<LayerRAM>()->getAsNormalizedDouble(
+            clickPos);
+
+    double nearClip = camera_.getNearPlaneDist();
+    double farClip = camera_.getFarPlaneDist();
+    double depthEye = farClip * nearClip / (farClip - depthNdc * (farClip - nearClip));
+    focusDepth_.set(depthEye);
+}
+
+void DepthOfField::process() {
+    int maxEvalCount = approximate_ ? viewCountApprox_.get() : viewCountExact_.get();
+    std::shared_ptr<const Image> img = inport_.getData();
+    size2_t dim = img->getDimensions();
+    TextureUnitContainer cont;
+
+    if (evalCount_ >= maxEvalCount) {
+        // Catch final network evaluation caused by camera resetting
+        evalCount_ = 0;
+        return;
+    } else if (evalCount_ == 0) {
+        setupRecursion(dim, maxEvalCount, img);
+    }
+
+    SkewedPerspectiveCamera* camera = dynamic_cast<SkewedPerspectiveCamera*>(&camera_.get());
+    double focusDepth = calculateFocusDepth();
+
+    if (approximate_) {
+        vec2 cameraPos = calculatePeripheralCameraPos(evalCount_, maxEvalCount);
+        double nearClip = camera_.getNearPlaneDist();
+        double farClip = camera_.getFarPlaneDist();
+        double fovy = glm::radians(camera->getFovy());
+        if (useComputeShaders_) {
+            warpToLightfieldGPU(cont, nearClip, farClip, fovy, focusDepth, dim, cameraPos);
+        } else {
+            warpToLightfieldCPU(img, nearClip, farClip, fovy, focusDepth, dim, cameraPos);
+        }
+    } else {
+        addToAccumulationBuffer(img, cont);
+    }
+
+    if (evalCount_ < maxEvalCount - 1) {
+        // Prepare camera for rendering again
+        if (!approximate_) {
+            nextOutImg_->copyRepresentationsTo(prevOutImg_.get());
+        }
+        moveCamera(camera, maxEvalCount, focusDepth);
+        evalCount_++;
+    } else {
+        // Set output
+        if (approximate_) {
+            synthesizeLightfield(cont);
+        } else {
+            outport_.setData(nextOutImg_);
+        }
+
+        // Reset camera
+        evalCount_ = (ogType_ == SkewedPerspectiveCamera::classIdentifier) ? 0 : maxEvalCount;
+        camera_.setCamera(std::unique_ptr<Camera>(ogCamera_->clone()));
+    }
+}
+
+void DepthOfField::setupRecursion(size2_t dim, int maxEvalCount, std::shared_ptr<const Image> img) {
+    ogCamera_.reset(camera_.get().clone());
+    ogType_ = camera_.get().getClassIdentifier();
+    if (ogType_ != PerspectiveCamera::classIdentifier) {
+        LogWarn("Intended for use with Perspective Camera. Unexpected behavior may follow.");
+    }
+    camera_.setCamera(SkewedPerspectiveCamera::classIdentifier);
+
+    if (approximate_) {
+        // Prepare light field
+        dimLightField_ = size3_t(dim.x, dim.y, viewCountApprox_.get() + simViewCountApprox_.get());
+        lightField_ = std::make_shared<Volume>(dimLightField_, DataFormat<vec4>::get());
```

Comment:
This creates other issues down the line, particularly when using this data in shaders. If it's just a matter of preference I'd rather leave it as it is.

Reply:
What kind of issue with shaders does it create?

Reply:
I couldn't easily convert it to a VolumeGL and since I don't currently have access to a machine which supports compute shaders, I feel uncertain about making such big changes.